cpp/src/arrow/flight/transport/grpc/util_internal.cc

```diff
@@ -20,7 +20,9 @@
 #include <cstdlib>
 #include <map>
 #include <memory>
+#include <optional>
 #include <string>
+#include "arrow/util/string_builder.h"
```

Comment:
Move this with other Arrow includes below.

Reply:
`clangd` is frustratingly inconsistent about where it places includes...